core/test-output.ts

```diff
@@ -0,0 +1,61 @@
+import { ITest } from "./_interfaces/test.i";
+import { MatchError } from "./errors/match-error";
+
+export class TestOutput {
+
+    private _outStream: NodeJS.WritableStream;
+
+    constructor (outStream: NodeJS.WritableStream) {
+        this._outStream = outStream;
+    }
+
+    private _writeOut(message: string): void {
+        this._outStream.write(message);
+    }
+
+    public emitVersion(): void {
+        this._writeOut("TAP version 13\n");
+    }
+
+    public emitPlan(testCount: number): void {
+        this._writeOut(`1..${testCount}\n`);
+    }
+
+    public emitPass(testId: number, test: ITest, testCaseArguments: Array<any>): void {
```

Comment:
@jamesrichford Potentially, but the `emitFail` function takes an error as well...